core/clustersmngr/cluster/delegating_cache.go

```diff
@@ -56,17 +61,18 @@ func (c *delegatingCacheCluster) makeCachingClient(leafClient client.Client) (cl
 
 	delegatingCache := newDelegatingCache(leafClient, cache, c.scheme)
 
-	delegatingClient, err := client.NewDelegatingClient(client.NewDelegatingClientInput{
-		CacheReader: delegatingCache,
-		Client:      leafClient,
-		// Non-exact field matches are not supported by the cache.
-		// https://github.com/kubernetes-sigs/controller-runtime/issues/612
-		// TODO: Research if we can change the way we query those events so we can enable the cache for it.
-		UncachedObjects:   []client.Object{&v1.Event{}},
-		CacheUnstructured: true,
+	cl, err := client.New(c.restConfig, client.Options{
+		Cache: &client.CacheOptions{
+			Reader: delegatingCache,
+			// Non-exact field matches are not supported by the cache.
+			// https://github.com/kubernetes-sigs/controller-runtime/issues/612
+			// TODO: Research if we can change the way we query those events so we can enable the cache for it.
```

Comment:
File an issue about it please, otherwise this comment will live there "forever".

Reply:
https://github.com/weaveworks/weave-gitops/issues/3814